src/libLookup/Lookup.cpp

```diff
@@ -2152,6 +2138,42 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
   cv_waitJoined.notify_all();
 }
 
+void Lookup::FindMissingMBsForLastNTxBlks(const uint32_t& num) {
+  LOG_MARKER();
+  uint64_t upperLimit =
+      m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum();
+  uint64_t lowerLimit = 1;
+
+  if (upperLimit > num) {
+    lowerLimit = upperLimit - num + 1;
```

Comment:
safemath here. will be required in the situation when we launch a new testnet to avoid underflow

Reply:
As discussed, check on `upperLimit and num` avoids underflow.